### ISSUE TITLE
Ask for admin credentials as arguments of `fractalctl start`

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#462]
Required
```
fractalctl start --admin-email admin@fractal.xy --admin-password 1234
```

It we can, let's also have:
```
fractalctl start                                       # (without arguments)
-> interactively asks for credentials
```

- [ ] Update docs

**Comment by mfranzon** [comment]
Following the uvicorn docs, the standard way to pass custom args to `app:main` is using a `.env` file with the ```--env-file``` flag.
https://www.uvicorn.org/settings/#settings

A possible solution is to have the `admin-email` and `admin-password` args declared in the `.fractal-server.env` file, ref #463 
I want to stress the fact that this SHOULD be just a temporary admin.

**Comment by tcompa** [comment]
@jluethi does the PR solve the issue that we discussed yesterday?
Would you rather make it such that there is no default password at all, and the server startup just fails if the env variable is missing?

Getting arguments from the command line is not an option, it turns out. There exist [workarounds](https://github.com/tiangolo/fastapi/issues/5822#issuecomment-1368506485), but we don't see a strong enough reason to mess with the server startup operation (which could lead to other problems, e.g. when using gunicorn instead of uvicorn).

Refs:
* https://github.com/encode/uvicorn/issues/168
* https://github.com/tiangolo/fastapi/issues/5822

**Comment by tcompa** [comment]
Also, notice the trade-off that we would be introducing with #463:
It's true, we don't have the first-user credentials hard-coded anywhere, but we now have them written in a file. If the step of password update is skipped, another potential issue appears (one should make sure that the env file always has very restrictive access).

TL;DR The server admin really should update the password during the startup, unless they are working on a private machine with no risk of external access.

**Comment by jluethi** [comment]
Hmm, interesting. Thanks for looking into this and exploring the options. I agree with not messing with the startup operation.

And thanks for adding the file-based option. It may be useful to someone wanting to run it in a specific way, having the user predefined based on a file. But as you say, having the password in an unencrypted file on disk does not sound like a very safe setup either. 

> TL;DR The server admin really should update the password during the startup, unless they are working on a private machine with no risk of external access.

Ok, sounds good. In this case, my proposal: 
We can add optional setting of passwords via a file. But there is still a standard password and the way we use it is the way it already exist: Default password is set and the documentation strongly encourages changing this password. In our examples, we just use the default password and change it. We can use a prompt for admin password in the reworked user registration step (at least in the version we share with other users as examples, not for testing purposes)

**Comment by tcompa** [comment]
OK, then the only additional changes in #463 are:
* Make variable names more explicit (e.g. `FRACTAL_ADMIN_USERNAME` --> `FRACTAL_DEFAULT_ADMIN_EMAIL` and `FRACTAL_ADMIN_PASSWORD` --> `FRACTAL_DEFAULT_ADMIN_PASSWORD`), to stress that those are only the defaults and not meant to be used later on.
* Clarify things in docstrings/docs.

**Comment by tcompa** [comment]
> We can use a prompt for admin password in the reworked user registration step (at least in the version we share with other users as examples, not for testing purposes)

I'm not sure I'm getting this point, could you make it more specific?

What we current have is that all `fractal user` commands require authentication with either one these options: `-u/-p` arguments, env variables (`FRACTAL_USER` and `FRACTAL_PASSWORD`), or .env file. And apart from `fractal user whoami`, all other `fractal user` commands require the user to also be a superuser.
The admin workflow should be:
* Startup the server, with whatever default admin credentials you like (that is, either the default ones or some credentials written in the .env file).
* Using those default admin credentials, run `fractal edit` to edit your own credentials.
* Use the new admin credentials for any other user operation, e.g. creating other users.

**Comment by jluethi** [comment]
> I'm not sure I'm getting this point, could you make it more specific?

What I meant was a flow similar to what you described below:

> Startup the server, with whatever default admin credentials you like (that is, either the default ones or some credentials written in the .env file).
> Using those default admin credentials, run fractal edit to edit your own credentials.

Here, in the example bash script we use to edit the user, ask for a user interactively to input a new (email &) password for the admin user. That way, when someone just executes our example scripts, those interactively ask for new credentials to be set.

> Use the new admin credentials for any other user operation, e.g. creating other users.

Same, now just with the admin credentials the user just set

**Comment by tcompa** [comment]
Got it, thanks.
(I thought it was a to-do change for fractal-client, but it's rather for fractal-demos scripts)
